app/dao/notifications_dao.py

```diff
@@ -176,22 +176,6 @@ def dao_create_notification(notification, notification_type, provider_identifier
                                             service_id=notification.service_id)
         db.session.add(template_stats)
 
-    update_count = db.session.query(ProviderStatistics).filter_by(
-        day=date.today(),
-        service_id=notification.service_id,
-        provider_id=provider.id
-    ).update({'unit_count': ProviderStatistics.unit_count + (
-        1 if notification_type == TEMPLATE_TYPE_EMAIL else get_sms_fragment_count(notification.content_char_count))})
-
-    if update_count == 0:
-        provider_stats = ProviderStatistics(
-            day=notification.created_at.date(),
-            service_id=notification.service_id,
-            provider_id=provider.id,
-            unit_count=1 if notification_type == TEMPLATE_TYPE_EMAIL else get_sms_fragment_count(
-                notification.content_char_count))
-        db.session.add(provider_stats)
-
     db.session.add(notification)
```

Comment:
Can we ensure that the notification update is the last thing that happens? Deadlocks now prey on my mind.